monkestation/code/modules/hydroponics/machines/genesischamber.dm

```diff
@@ -0,0 +1,255 @@
+
+	to_chat(user, span_notice("You collect [capacity] strange seeds from the chamber."))
+	capacity = 0
+	update_appearance()
+
+/obj/machinery/genesis_chamber/proc/calculate_time_remaining()
+	if(!on || !seed_timer_id)
+		time_remaining = max_cooldown
+		return
+
+	var/elapsed_time = world.time - generation_start_time
+	time_remaining = max(0, max_cooldown - elapsed_time)
+
+/obj/machinery/genesis_chamber/attack_hand_secondary(mob/living/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+
+	if(obj_flags & EMAGGED)
+		return
+
+	if(panel_open)
+		return
+
+	on = !on
+
+	if(on)
+		to_chat(user, span_notice("You turn on [src]."))
+		playsound(loc, 'sound/machines/chime.ogg', 40, TRUE)
+		say("Activating... Seeds will be produced in [DisplayTimeText(max_cooldown)].")
+		START_PROCESSING(SSprocessing, src)
+		generation_start_time = world.time
+		seed_timer_id = addtimer(CALLBACK(src, PROC_REF(generate_seed)), max_cooldown, TIMER_STOPPABLE | TIMER_UNIQUE | TIMER_DELETE_ME)
+	else
+		to_chat(user, span_notice("You turn off [src]."))
+		playsound(loc, 'sound/machines/click.ogg', 15, TRUE, -3)
+		say("Deactivating...")
+		STOP_PROCESSING(SSprocessing, src)
+		if(seed_timer_id)
+			deltimer(seed_timer_id)
+			seed_timer_id = null
+		time_remaining = max_cooldown
+
+	update_appearance()
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/machinery/genesis_chamber/process()
+	if(!on)
+		return PROCESS_KILL
+
+	calculate_time_remaining()
+
+	if(capacity >= max_capacity)
+		on = FALSE
+		say("Maximum capacity reached. Shutting down.")
+		playsound(src, 'sound/machines/ping.ogg', 50, TRUE)
+		update_appearance()
+		return PROCESS_KILL
+
+	return TRUE
+
+/obj/machinery/genesis_chamber/proc/generate_seed()
+	if(!on || capacity >= max_capacity)
+		seed_timer_id = null
+		return
+
+	capacity++
+	say("Seed generated. Current capacity: [capacity]/[max_capacity]")
+	playsound(src, 'sound/machines/synth_yes.ogg', 30, TRUE)
+
+	seed_timer_id = null
+	if(on && capacity < max_capacity)
+		generation_start_time = world.time
+		seed_timer_id = addtimer(CALLBACK(src, PROC_REF(generate_seed)), max_cooldown, TIMER_STOPPABLE | TIMER_UNIQUE | TIMER_DELETE_ME)
+
+	update_appearance()
+
+/obj/machinery/genesis_chamber/examine(mob/user)
+	. = ..()
+	if(!on)
+		. += "It is currently <b>Deactivated</b> and has <b>[capacity]</b> strange seeds stored."
+	else
+		. += "It is currently <b>Activated</b> and has <b>[capacity]</b> strange seeds stored."
+		calculate_time_remaining()
+		. += "It'll take <b>[DisplayTimeText(time_remaining)]</b> until another seed is created."
+	. += "The [src] can currently hold <b>[max_capacity] seeds</b> and produces each one every <b>[DisplayTimeText(max_cooldown)]</b>."
+
+/obj/machinery/genesis_chamber/update_overlays()
+	. = ..()
+	if (capacity >= 1)
+		icon_state = "genesis_chamber_seed_[on ? "on" : "off"]"
+		return ..()
+	icon_state = "genesis_chamber_[on ? "on" : "off"]"
+
+	return ..()
```

Comment:
`update_overlays` is for overlays. you want to override `update_icon_state`

also don't call `..()` twice